packages/graphql/lib/schema-builder/storages/type-metadata.storage.ts

```diff
@@ -394,6 +378,41 @@ export class TypeMetadataStorageHost {
     }
   }
 
+  private findModelFieldMetadata(
+    item: FieldResolverMetadata,
+  ): [Function, ClassMetadata, PropertyMetadata | undefined] {
+    const objectTypeRef = this.metadataByTargetCollection
+      .get(item.target)
+      .resolver.typeFn();
+    const getTypeMetadata = (objectTypeRef: any) =>
+      this.metadataByTargetCollection.get(objectTypeRef).objectType ||
+      this.metadataByTargetCollection.get(objectTypeRef).interface;
+    for (
+      let _objectTypeRef = objectTypeRef;
+      _objectTypeRef?.prototype;
+      _objectTypeRef = Object.getPrototypeOf(_objectTypeRef)
+    ) {
+      const possibleTypeMetadata = getTypeMetadata(_objectTypeRef);
```

Comment:
What if `IngredientBase` wasn't annotated with the `ObjectType` decorator? Would this method throw an exception? Is that a breaking change, or was that the case before too?

Reply:
@kamilmysliwiec Good point. Prior to this PR it wouldn't add the field without the decorator but it also wouldn't throw as long as the field was defined in the `Resolver`. I've added a case to ensure we don't throw when a) the field is defined in a base class that doesn't have the `ObjectType` decorator and b) there is a `Field()` defined for that field in the resolver.